src/Meltor.php

```diff
@@ -0,0 +1,872 @@
+    {
+        return collect($connection->getDoctrineSchemaManager()->listTableIndexes($tableName))->filter->isPrimary()->all()['primary'] ?? null;
+    }
+
+    /**
+     * Returns a migration template.
+     *
+     * Exists to help refactor the handling of migration templates.
+     *
+     * @param string $name
+     * @param null $default
+     *
+     * @return string|null
+     */
+    public function getMigrationTemplate(string $name, $default = null): ?string
+    {
+        $value = config(sprintf('meltor-templates.%s', $name), $default);
+
+        return is_callable($value) ? $value() : $value;
+    }
+
+    /**
+     * Returns a laravel migration command to alter or create a field.
+     *
+     * @param string $method
+     * @param string|null $columnsString
+     * @param string|null $keyName
+     * @return string
+     */
+    protected function compileColumnMigration(string $method, string $columnsString = null, string $keyName = null): string
+    {
+        $template = $this->getMigrationTemplate('column');
+
+        if ($keyName) {
+            return sprintf($template, sprintf('%s(%s, \'%s\')', $method, $columnsString, $keyName));
+        }
+
+        if ($columnsString) {
+            return sprintf($template, sprintf('%s(%s)', $method, $columnsString));
+        }
+
+        return sprintf($template, $method);
+    }
+
+    /**
+     * Returns a string for one or more fields for a migration command.
+     *
+     * @param array $columns
+     * @return string
+     */
+    protected function compileColumnsString(array $columns): string
+    {
+        if (count($columns) > 1) {
+            return sprintf('[\'%s\']', implode('\', \'', $columns));
+        } else {
+            return sprintf('\'%s\'', $columns[0]);
+        }
+    }
+
+    /**
+     * Returns a raw DB statement for MySQL COLUMN_TYPES that are not supported by Laravel.
+     *
+     * @param string $table
+     * @param string $name
+     * @param string $type
+     * @return string
+     */
+    public function generateRawColumnMigration(string $table, string $name, string $type): string
+    {
+        return sprintf($this->getMigrationTemplate('columnRaw'), $table, $name, $type);
+    }
+
+    /**
+     * Returns the content of a Laravel migration file.
+     *
+     * @param array $tableMigrations
+     * @param string $comment
+     * @param array $constraintMigrations
+     *
+     * @return string
+     */
+    public function generateMigration(array $tableMigrations, string $comment, array $constraintMigrations = [], array $rawExceptions = []): string
+    {
+        $tableMigrationCode = [];
+
+        foreach ($tableMigrations as $tableName => $columnMigrations) {
+            $columns              = implode("\n", $columnMigrations['laravel']);
+            $columnsRaw           = count($columnMigrations['raw']) ? "\n" . implode("\n", $columnMigrations['raw']) : '';
```

Comment:
```suggestion
            $columnsRaw           =  collect($columnMigrations['raw'])
            ->whenNotEmpty(fn($columns) => $columns->prepend(''))
            ->implode("\n")
```
Not really much shorter or nicer. The extra newline doubles the code.